src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1667,14 +1667,20 @@ class RGWSpec(ServiceSpec):
 
     """
     def __init__(self,
-                 rgw_realm,  # type: str
-                 rgw_zone,  # type: str
+                 rgw_realm=None,  # type: Optional[str]
+                 rgw_zone=None,  # type: Optional[str]
+                 service_id=None, # type: Optional[str]
                  placement=None,
                  service_type='rgw',
                  rgw_frontend_port=None,  # type: Optional[int]
                  ):
         assert service_type == 'rgw'
-        super(RGWSpec, self).__init__('rgw', service_id=rgw_realm+'.'+rgw_zone, placement=placement)
+        if service_id:
+            (rgw_realm, rgw_zone) = service_id.split('.', 1)
```

Comment:
Not sure what this is doing?

Reply:
it allows you to pass service_id='realm.zone' and the ctor will still populate realm_id and zone_id.

Reply:
oh, right the super() was moved below that. marking as resovled then